opds_import.py

```diff
@@ -556,7 +552,40 @@ def _data_detail_for_feedparser_entry(cls, entry, data_source):
         if title == OPDSFeed.NO_TITLE:
             title = None
         subtitle = entry.get('schema_alternativeheadline', None)
-        
+
+        # Generally speaking, a data source will provide either
+        # metadata (e.g. the Simplified metadata wrangler) or both
+        # metadata and circulation data (e.g. a publisher's ODL feed).
+        #
+        # However there is at least one case (the Simplified
+        # open-access content server) where one server provides
+        # circulation data from a _different_ data source
+        # (e.g. Project Gutenberg).
+        #
+        # In this case we want the data source of the LicensePool to
+        # be Project Gutenberg, but the data source of the pool's
+        # presentation to be the open-access content server.
+        #
+        # The open-access content server uses a
+        # <bibframe:distribution> tag to keep track of which data
+        # source provides the circulation data.
+        circulation_data_source = metadata_data_source
+        circulation_data_source_tag = entry.get('bibframe_distribution')
+        if circulation_data_source_tag:
+            circulation_data_source_name = circulation_data_source_tag.get(
+                'bibframe:providername'
+            )
+            if circulation_data_source_name:
+                _db = Session.object_session(metadata_data_source)
+                circulation_data_source = DataSource.lookup(
+                    _db, circulation_data_source_name
+                )
+                if not circulation_data_source:
+                    raise ValueError(
+                        "Unrecognized circulation data source: %s" % (
+                            circulation_data_source_name
+                        )
+                    )
```

Comment:
If a data source is added to the content server, but the circ manager doesn't know about it yet, this code stops a crash.